core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -866,26 +872,30 @@ public PlanOptimizers(
         builder.add(new StatsRecordingPlanOptimizer(
                 optimizerStats,
                 new PredicatePushDown(metadata, typeOperators, typeAnalyzer, true, false)));
-        builder.add(simplifyOptimizer); // Should be always run after PredicatePushDown
         builder.add(new IterativeOptimizer(
                 metadata,
                 ruleStats,
                 statsCalculator,
                 costCalculator,
-                ImmutableSet.of(new RemoveRedundantTableScanPredicate(metadata, typeOperators))));
+                ImmutableSet.<Rule<?>>builder()
+                        .addAll(simplifyOptimizerRules) // Should be always run after PredicatePushDown
+                        .add(new RemoveRedundantTableScanPredicate(metadata, typeOperators))
```

Comment:
apparently this one should also be run after every PPD

Reply:
out of scope for this PR